src/sources/aws_ecs_metrics/mod.rs

```diff
@@ -144,41 +145,56 @@ async fn aws_ecs_metrics(
 
                         let byte_size = body.len();
 
+                        emit!(&HttpBytesReceived {
+                            byte_size,
+                            protocol: "http",
+                            http_path: uri.path(),
+                        });
+
                         match parser::parse(body.as_ref(), namespace.clone()) {
                             Ok(metrics) => {
-                                emit!(&AwsEcsMetricsReceived {
+                                let count = metrics.len();
+                                emit!(&HttpEventsReceived {
                                     byte_size,
```

Comment:
I think this `byte_size` is actually wrong too. It should be using `ByteSizeOf::size_of` to calculate the in-memory byte count.